plugins/workflow/category/src/Extension/Category.php

```diff
@@ -0,0 +1,259 @@
+     * The form event.
+     *
+     * @param   PrepareFormEvent  $event  The event
+     *
+     * @since   DEPLOY_VERSION
+     */
+    public function onContentPrepareForm(PrepareFormEvent $event)
+    {
+        $form = $event->getForm();
+        $data = $event->getData();
+
+        $context = $form->getName();
+
+        // Extend the transition form
+        if ($context === 'com_workflow.transition') {
+            $this->enhanceWorkflowTransitionForm($form, $data);
+
+            return;
+        }
+
+        if ($context === 'com_content.article') {
+            if (empty($data && $data->id)) {
+                return;
+            }
+            $this->disableCategoryField($form, $data);
+        }
+    }
+
+    /**
+     * Manipulate the generic list view
+     *
+     * @param   DisplayEvent  $event
+     *
+     * @return  void
+     *
+     * @since   4.0.0
+     */
+    public function onAfterDisplay(DisplayEvent $event)
+    {
+        if (!$this->getApplication()->isClient('administrator')) {
+            return;
+        }
+
+        $js = "
+            document.addEventListener('DOMContentLoaded', function()
+            {
+                var dropdown = document.getElementById('toolbar-status-group');
+                if (!dropdown){
+                    return;
+                }
+                var batchButton = document.getElementById('status-group-children-batch');
+                if (batchButton){
+                    batchButton.addEventListener('click', function()
+                    {
+                        var observer = new MutationObserver(function(mutations, observer) {
+                            var categorySelector = document.getElementById('batch-category-id');
+                            if (categorySelector) {
+                                categorySelector.disabled = true;
+                                observer.disconnect();
+                            }
+                        });
+
+                        observer.observe(document.body, { childList: true, subtree: true });
+                    });
+                }
+            });
+        ";
+
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
+    }
+
+    /**
+     * Check if the current plugin should execute workflow related activities
+     *
+     * @param   string  $context
+     *
+     * @return   boolean
+     *
+     * @since   DEPLOY_VERSION
+     */
+    protected function isSupported($context)
+    {
+        return in_array($context, ['com_workflow.transition', 'com_content.article']);
+    }
+
+    /**
+     * Disable the category field in the article form.
+     *
+     * @param   Form      $form  The form
+     * @param   \stdClass    $data  The data
+     *
+     * @return  void
+     *
+     * @since   DEPLOY_VERSION
+     */
+    protected function disableCategoryField(Form $form, $data)
+    {
+        // Get the current category ID value
+        $catid = $form->getValue('catid');
```

Comment:
not every category-field is called "catid", you can get the correct name like: https://github.com/joomla/joomla-cms/blob/5.3-dev/plugins/workflow/featuring/src/Extension/Featuring.php#L136